TestIssue2904 fails

https://semaphoreci.com/coreos/etcd/branches/pull-request-4277/builds/1

```
=== RUN   TestIssue2904
2016-01-25 13:55:29.305703 C | etcdserver: first index of committed entry[18] should <= appliedi[0] + 1
panic: first index of committed entry[18] should <= appliedi[0] + 1

goroutine 102735 [running]:
github.com/coreos/etcd/Godeps/_workspace/src/github.com/coreos/pkg/capnslog.(*PackageLogger).Panicf(0xc82018c280, 0xfed4e0, 0x3d, 0xc8201a7920, 0x2, 0x2)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/coreos/pkg/capnslog/pkg_logger.go:73 +0x191
github.com/coreos/etcd/etcdserver.(*EtcdServer).applyEntries(0xc8327701a0, 0xc832768180, 0xc8201a7ef8)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:664 +0x20d
github.com/coreos/etcd/etcdserver.(*EtcdServer).applyAll(0xc8327701a0, 0xc832768180, 0xc8201a7ef8)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:590 +0x5b
github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier.func1.1(0xc820099b10, 0x1, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:501 +0x61
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier.func1
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:523 +0x42a

goroutine 1 [chan receive]:
testing.RunTests(0x107c1a0, 0x13af660, 0x43, 0x43, 0x1)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc821cf5ef8, 0x0)
	/usr/local/go/src/testing/testing.go:494 +0x70
github.com/coreos/etcd/integration.TestMain(0xc821cf5ef8)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/main_test.go:15 +0x21
main.main()
	github.com/coreos/etcd/integration/_test/_testmain.go:190 +0x113

goroutine 17 [syscall, 6 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 13 [chan receive]:
github.com/coreos/etcd/Godeps/_workspace/src/github.com/golang/glog.(*loggingT).flushDaemon(0x13b4840)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x67
created by github.com/coreos/etcd/Godeps/_workspace/src/github.com/golang/glog.init.1
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x297

goroutine 20 [chan receive]:
github.com/coreos/etcd/pkg/logutil.(*MergeLogger).outputLoop(0xc820168320)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:174 +0x7c
created by github.com/coreos/etcd/pkg/logutil.NewMergeLogger
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:92 +0xc0

goroutine 16 [chan receive]:
github.com/coreos/etcd/pkg/logutil.(*MergeLogger).outputLoop(0xc820155660)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:174 +0x7c
created by github.com/coreos/etcd/pkg/logutil.NewMergeLogger
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:92 +0xc0

goroutine 102789 [runnable]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier.func1.1(0xc8200997f8, 0xb, 0x19, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:509 +0x172
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier.func1
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:517 +0x2bf

goroutine 100920 [IO wait]:
net.runtime_pollWait(0x7f8df5e7c1d0, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8225a0a70, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8225a0a70, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8225a0a10, 0x0, 0x7f8df9d43c50, 0xc821d48c80)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc829682060, 0xc8201a1b40, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc829682060, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:264 +0x3d
net/http/httptest.(*historyListener).Accept(0xc829623800, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/httptest/server.go:48 +0x63
net/http.(*Server).Serve(0xc831719980, 0x7f8df9d45fe8, 0xc829623800, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1887 +0xb3
created by net/http/httptest.(*Server).Start
	/usr/local/go/src/net/http/httptest/server.go:109 +0x380

goroutine 101011 [IO wait]:
net.runtime_pollWait(0x7f8df9d6f180, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc828b36060, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc828b36060, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc828b36000, 0xc821e5d000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8296822e0, 0xc821e5d000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
github.com/coreos/etcd/pkg/transport.timeoutConn.Read(0x7f8df9d4a5a8, 0xc8296822e0, 0x0, 0x0, 0xc821e5d000, 0x1000, 0x1000, 0x5ca7c6, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/transport/timeout_conn.go:43 +0x1b8
github.com/coreos/etcd/pkg/transport.(*timeoutConn).Read(0xc8326f53e0, 0xc821e5d000, 0x1000, 0x1000, 0xc8318b67d8, 0x0, 0x0)
	<autogenerated>:42 +0xc8
net/http.noteEOFReader.Read(0x7f8df9d4a920, 0xc8326f53e0, 0xc820c84f78, 0xc821e5d000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8326f5400, 0xc821e5d000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc82a0e0ba0)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82a0e0ba0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc820c84f20)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 100917 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).purgeFile(0xc8327701a0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:444 +0x4bf
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).Start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:410 +0x88

goroutine 100859 [runnable]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier.func1(0xc821cbe300, 0xc8327701a0, 0xc832768180)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:512 +0x495
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:530 +0xa4

goroutine 102763 [select]:
github.com/coreos/etcd/Godeps/_workspace/src/github.com/xiang90/probing.(*prober).AddHTTP.func1(0xc82266d5c0, 0xc82018b2b0, 0x1, 0x1, 0xc8320a3380, 0xc82266d580)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/xiang90/probing/prober.go:55 +0x4cd
created by github.com/coreos/etcd/Godeps/_workspace/src/github.com/xiang90/probing.(*prober).AddHTTP
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/xiang90/probing/prober.go:85 +0x239

goroutine 102804 [sleep]:
time.Sleep(0x3b9aca00)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/coreos/etcd/etcdserver.(*EtcdServer).stopWithDelay(0xc820158000, 0x3b9aca00, 0x7f8df9d3f028, 0xc82100e0e0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:707 +0x21
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).applyEntries
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:672 +0x34f

goroutine 102745 [IO wait]:
net.runtime_pollWait(0x7f8df9d4f2e8, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822662d80, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822662d80, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc822662d20, 0x0, 0x7f8df9d43c50, 0xc8320a3d20)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc824b96250, 0xc8320a4080, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc824b96250, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:264 +0x3d
net/http/httptest.(*historyListener).Accept(0xc8205738f0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/httptest/server.go:48 +0x63
net/http.(*Server).Serve(0xc822abfce0, 0x7f8df9d45fe8, 0xc8205738f0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1887 +0xb3
created by net/http/httptest.(*Server).Start
	/usr/local/go/src/net/http/httptest/server.go:109 +0x380

goroutine 102753 [IO wait]:
net.runtime_pollWait(0x7f8df9d4ece8, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822663790, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822663790, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc822663730, 0xc832f91000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc824b96390, 0xc832f91000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.(*liveSwitchReader).Read(0xc821d4d2d8, 0xc832f91000, 0x1000, 0x1000, 0xc83179a000, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:219 +0xa4
io.(*LimitedReader).Read(0xc8320a3d00, 0xc832f91000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:427 +0xbd
bufio.(*Reader).fill(0xc8318b74a0)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8318b74a0, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*conn).readRequest(0xc821d4d290, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:629 +0x2bb
net/http.(*conn).serve(0xc821d4d290)
	/usr/local/go/src/net/http/server.go:1319 +0x727
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1910 +0x3f6

goroutine 102700 [IO wait]:
net.runtime_pollWait(0x7f8df9d6f900, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822663020, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822663020, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc822662fc0, 0xc82025f000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8296823e8, 0xc82025f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
github.com/coreos/etcd/pkg/transport.timeoutConn.Read(0x7f8df9d4a5a8, 0xc8296823e8, 0x0, 0x0, 0xc82025f000, 0x1000, 0x1000, 0x5ca7c6, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/transport/timeout_conn.go:43 +0x1b8
github.com/coreos/etcd/pkg/transport.(*timeoutConn).Read(0xc821f5fcc0, 0xc82025f000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
	<autogenerated>:42 +0xc8
net/http.noteEOFReader.Read(0x7f8df9d4a920, 0xc821f5fcc0, 0xc821d4ce18, 0xc82025f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc821f5fce0, 0xc82025f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc822aa1020)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc822aa1020, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc821d4cdc0)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 102725 [IO wait]:
net.runtime_pollWait(0x7f8df9d4f468, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822660ca0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822660ca0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc822660c40, 0xc821d39000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200304d8, 0xc821d39000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
github.com/coreos/etcd/pkg/transport.timeoutConn.Read(0x7f8df9d4a5a8, 0xc8200304d8, 0x0, 0x0, 0xc821d39000, 0x1000, 0x1000, 0x5ca7c6, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/transport/timeout_conn.go:43 +0x1b8
github.com/coreos/etcd/pkg/transport.(*timeoutConn).Read(0xc82265e7e0, 0xc821d39000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
	<autogenerated>:42 +0xc8
net/http.noteEOFReader.Read(0x7f8df9d4a920, 0xc82265e7e0, 0xc820ffc268, 0xc821d39000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc82265e800, 0xc821d39000, 0x1000, 0x1000, 0xc824b96398, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc8225a5a40)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8225a5a40, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc820ffc210)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 100918 [select]:
github.com/coreos/etcd/etcdserver.monitorFileDescriptor(0xc8317197a0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/metrics.go:81 +0x536
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).Start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:411 +0xb1

goroutine 102705 [select]:
github.com/coreos/etcd/rafthttp.(*streamWriter).run(0xc8226663f0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:140 +0xe33
created by github.com/coreos/etcd/rafthttp.startStreamWriter
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:127 +0x1c8

goroutine 100914 [runnable]:
sync.(*Mutex).Unlock(0xc8325047e0)
	/usr/local/go/src/sync/mutex.go:99
github.com/coreos/etcd/raft.(*MemoryStorage).FirstIndex(0xc8325047e0, 0x1, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/storage.go:145 +0xc6
github.com/coreos/etcd/raft.(*raftLog).firstIndex(0xc8225a1960, 0xc820099b10)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/log.go:172 +0x7b
github.com/coreos/etcd/raft.(*raftLog).mustCheckOutOfBounds(0xc8225a1960, 0x12, 0x13, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/log.go:340 +0x1ac
github.com/coreos/etcd/raft.(*raftLog).slice(0xc8225a1960, 0x12, 0x13, 0xffffffffffffffff, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/log.go:298 +0x79
github.com/coreos/etcd/raft.(*raftLog).nextEnts(0xc8225a1960, 0x0, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/log.go:142 +0x9f
github.com/coreos/etcd/raft.newReady(0xc8325048c0, 0xc82035b230, 0x2, 0xc7b14cb1acd6356c, 0x11, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:450 +0x6f
github.com/coreos/etcd/raft.(*node).run(0xc833962dc0, 0xc8325048c0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:252 +0x5cf
created by github.com/coreos/etcd/raft.StartNode
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:178 +0x73d

goroutine 102743 [select]:
github.com/coreos/etcd/etcdserver.monitorFileDescriptor(0xc822abfc20)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/metrics.go:81 +0x536
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).Start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:411 +0xb1

goroutine 102756 [select]:
github.com/coreos/etcd/rafthttp.(*pipeline).handle(0xc822a53680)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:91 +0x615
created by github.com/coreos/etcd/rafthttp.newPipeline
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:77 +0x205

goroutine 102742 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).purgeFile(0xc820158000)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:444 +0x4bf
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).Start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:410 +0x88

goroutine 102751 [IO wait]:
net.runtime_pollWait(0x7f8df9d77688, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822662df0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822662df0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc822662d90, 0x0, 0x7f8df9d43c50, 0xc8320a3c20)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc824b96260, 0xc800000001, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc824b96260, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:264 +0x3d
net/http/httptest.(*historyListener).Accept(0xc83380eb10, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/httptest/server.go:48 +0x63
net/http.(*Server).Serve(0xc822abfec0, 0x7f8df9d45fe8, 0xc83380eb10, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1887 +0xb3
created by net/http/httptest.(*Server).Start
	/usr/local/go/src/net/http/httptest/server.go:109 +0x380

goroutine 100921 [select]:
github.com/coreos/etcd/etcdserver/etcdhttp.capabilityLoop(0xc8327701a0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/etcdhttp/capability.go:68 +0x1ad
created by github.com/coreos/etcd/etcdserver/etcdhttp.NewClientHandler
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/etcdhttp/client.go:63 +0x6b

goroutine 101012 [select]:
net/http.(*persistConn).writeLoop(0xc820c84f20)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 102760 [select]:
github.com/coreos/etcd/rafthttp.(*streamReader).run(0xc821ae3e60)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:285 +0x1c8
created by github.com/coreos/etcd/rafthttp.startStreamReader
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:261 +0x1e5

goroutine 102757 [select]:
github.com/coreos/etcd/rafthttp.(*pipeline).handle(0xc822a53680)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:91 +0x615
created by github.com/coreos/etcd/rafthttp.newPipeline
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:77 +0x205

goroutine 100858 [select]:
github.com/coreos/etcd/etcdserver.(*raftNode).start.func1(0xc8327701a0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/raft.go:139 +0xbf3
created by github.com/coreos/etcd/etcdserver.(*raftNode).start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/raft.go:204 +0x116

goroutine 102759 [select]:
github.com/coreos/etcd/rafthttp.startPeer.func1(0xc821ae3dd0, 0x7f8df9d45d78, 0xc820158000, 0x7f8df9d4a568, 0xc82266d540)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/peer.go:146 +0x29e
created by github.com/coreos/etcd/rafthttp.startPeer
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/peer.go:155 +0x854

goroutine 102761 [select]:
github.com/coreos/etcd/rafthttp.(*streamReader).run(0xc821ae3ef0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:285 +0x1c8
created by github.com/coreos/etcd/rafthttp.startStreamReader
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:261 +0x1e5

goroutine 102755 [select]:
github.com/coreos/etcd/rafthttp.(*pipeline).handle(0xc822a53680)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:91 +0x615
created by github.com/coreos/etcd/rafthttp.newPipeline
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:77 +0x205

goroutine 102780 [IO wait]:
net.runtime_pollWait(0x7f8df9d4eda8, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822663db0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822663db0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc822663d50, 0xc821009000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc824b963b8, 0xc821009000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
github.com/coreos/etcd/pkg/transport.timeoutConn.Read(0x7f8df9d4a5a8, 0xc824b963b8, 0x0, 0x0, 0xc821009000, 0x1000, 0x1000, 0x5ca7c6, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/transport/timeout_conn.go:43 +0x1b8
github.com/coreos/etcd/pkg/transport.(*timeoutConn).Read(0xc82260e720, 0xc821009000, 0x1000, 0x1000, 0xc821cdadd8, 0x0, 0x0)
	<autogenerated>:42 +0xc8
net/http.noteEOFReader.Read(0x7f8df9d4a920, 0xc82260e720, 0xc821d4d708, 0xc821009000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc82260e760, 0xc821009000, 0x1000, 0x1000, 0xc821f2db80, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc821cdac60)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc821cdac60, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc821d4d6b0)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 102741 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).Do(0xc820158000, 0x7f8df9d4a480, 0xc822aa13e0, 0x9352791165950001, 0xea9788, 0x3, 0xc833954f00, 0x26, 0xc8317d8690, 0x49, ...)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:744 +0xbc4
github.com/coreos/etcd/etcdserver.(*EtcdServer).publish(0xc820158000, 0x135f1b400)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:921 +0x341
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).Start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:409 +0x66

goroutine 100919 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).monitorVersions(0xc8327701a0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:1219 +0x43f
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).Start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:412 +0xd3

goroutine 102767 [select]:
net/http.(*persistConn).writeLoop(0xc821d4d550)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 102703 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier.func1(0xc822aa12c0, 0xc820158000, 0xc82266d380)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:512 +0x495
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).startApplier
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:530 +0xa4

goroutine 102701 [select]:
net/http.(*persistConn).writeLoop(0xc821d4cdc0)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 102739 [select]:
github.com/coreos/etcd/raft.(*node).run(0xc832098640, 0xc8205f8540)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:275 +0x1354
created by github.com/coreos/etcd/raft.RestartNode
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:190 +0x2e3

goroutine 102768 [IO wait]:
net.runtime_pollWait(0x7f8df9d44b28, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822666610, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822666610, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8226665b0, 0xc832708000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc829682430, 0xc832708000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.(*liveSwitchReader).Read(0xc822668888, 0xc832708000, 0x1000, 0x1000, 0xc822611000, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:219 +0xa4
io.(*LimitedReader).Read(0xc821d488e0, 0xc832708000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:427 +0xbd
bufio.(*Reader).fill(0xc82a2f9ec0)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82a2f9ec0, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*conn).readRequest(0xc822668840, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:629 +0x2bb
net/http.(*conn).serve(0xc822668840)
	/usr/local/go/src/net/http/server.go:1319 +0x727
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1910 +0x3f6

goroutine 102762 [select]:
github.com/coreos/etcd/rafthttp.startPeer.func2(0xc821ae3dd0, 0xc8339553b0, 0x7f8df9d45d78, 0xc820158000, 0xc833955380, 0xc82018b290, 0xc821ae3ef0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/peer.go:162 +0xba9
created by github.com/coreos/etcd/rafthttp.startPeer
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/peer.go:202 +0xa1a

goroutine 102740 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).run(0xc820158000)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:571 +0x612
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:433 +0x49a

goroutine 101003 [IO wait]:
net.runtime_pollWait(0x7f8df9d70c10, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820fc64c0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820fc64c0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820fc6460, 0xc831892000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc824b96190, 0xc831892000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.(*liveSwitchReader).Read(0xc822e00f68, 0xc831892000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:219 +0xa4
io.(*LimitedReader).Read(0xc83270fb00, 0xc831892000, 0x1000, 0x1000, 0x11, 0x0, 0x0)
	/usr/local/go/src/io/io.go:427 +0xbd
bufio.(*Reader).fill(0xc83177b9e0)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).ReadSlice(0xc83177b9e0, 0xc81fb4510a, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:328 +0x21a
bufio.(*Reader).ReadLine(0xc83177b9e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:357 +0x53
net/textproto.(*Reader).readLineSlice(0xc828090ff0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/textproto/reader.go:55 +0x81
net/textproto.(*Reader).ReadLine(0xc828090ff0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/textproto/reader.go:36 +0x40
net/http.ReadRequest(0xc83177b9e0, 0xc82975cb60, 0x0, 0x0)
	/usr/local/go/src/net/http/request.go:653 +0xb6
net/http.(*conn).readRequest(0xc822e00f20, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:633 +0x32f
net/http.(*conn).serve(0xc822e00f20)
	/usr/local/go/src/net/http/server.go:1319 +0x727
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1910 +0x3f6

goroutine 102702 [select]:
github.com/coreos/etcd/etcdserver.(*raftNode).start.func1(0xc820158000)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/raft.go:139 +0xbf3
created by github.com/coreos/etcd/etcdserver.(*raftNode).start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/raft.go:204 +0x116

goroutine 102786 [IO wait]:
net.runtime_pollWait(0x7f8df9d4f768, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8226666f0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8226666f0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc822666690, 0xc820203000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc829682440, 0xc820203000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.(*liveSwitchReader).Read(0xc822668a98, 0xc820203000, 0x1000, 0x1000, 0xc82729e000, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:219 +0xa4
io.(*LimitedReader).Read(0xc821d48c60, 0xc820203000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:427 +0xbd
bufio.(*Reader).fill(0xc83395ad20)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc83395ad20, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*conn).readRequest(0xc822668a50, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:629 +0x2bb
net/http.(*conn).serve(0xc822668a50)
	/usr/local/go/src/net/http/server.go:1319 +0x727
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1910 +0x3f6

goroutine 102754 [select]:
github.com/coreos/etcd/rafthttp.(*streamWriter).run(0xc822666460)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:140 +0xe33
created by github.com/coreos/etcd/rafthttp.startStreamWriter
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:127 +0x1c8

goroutine 102764 [select]:
github.com/coreos/etcd/rafthttp.monitorProbingStatus(0x7f8df9d4e678, 0xc82266d580, 0xc82018b2a0, 0x10)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/probing_status.go:48 +0x4df
created by github.com/coreos/etcd/rafthttp.addPeerToProber
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/probing_status.go:42 +0x37a

goroutine 102752 [IO wait]:
net.runtime_pollWait(0x7f8df9d760c8, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822663720, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822663720, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8226636c0, 0xc8219c1000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc824b96388, 0xc8219c1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.(*liveSwitchReader).Read(0xc821d4d228, 0xc8219c1000, 0x1000, 0x1000, 0xc822a56000, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:219 +0xa4
io.(*LimitedReader).Read(0xc8320a3c80, 0xc8219c1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:427 +0xbd
bufio.(*Reader).fill(0xc8318b7440)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8318b7440, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*conn).readRequest(0xc821d4d1e0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:629 +0x2bb
net/http.(*conn).serve(0xc821d4d1e0)
	/usr/local/go/src/net/http/server.go:1319 +0x727
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1910 +0x3f6

goroutine 100832 [chan receive]:
github.com/coreos/etcd/integration.TestIssue2904(0xc8316be480)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:315 +0x495
testing.tRunner(0xc8316be480, 0x13af7c8)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d

goroutine 102766 [IO wait]:
net.runtime_pollWait(0x7f8df5e7ca10, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc822663b10, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822663b10, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc822663ab0, 0xc8202a5000, 0x1000, 0x1000, 0x0, 0x7f8df9d3f050, 0xc8200141d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc829682420, 0xc8202a5000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
github.com/coreos/etcd/pkg/transport.timeoutConn.Read(0x7f8df9d4a5a8, 0xc829682420, 0x0, 0x0, 0xc8202a5000, 0x1000, 0x1000, 0x5ca7c6, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/transport/timeout_conn.go:43 +0x1b8
github.com/coreos/etcd/pkg/transport.(*timeoutConn).Read(0xc821d48800, 0xc8202a5000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
	<autogenerated>:42 +0xc8
net/http.noteEOFReader.Read(0x7f8df9d4a920, 0xc821d48800, 0xc821d4d5a8, 0xc8202a5000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc821d48820, 0xc8202a5000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc821f38720)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc821f38720, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc821d4d550)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 102744 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).monitorVersions(0xc820158000)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:1219 +0x43f
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).Start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:412 +0xd3

goroutine 100926 [IO wait]:
net.runtime_pollWait(0x7f8df9d75800, 0x72, 0xc8200141d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8225a0b50, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8225a0b50, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8225a0af0, 0x0, 0x7f8df9d43c50, 0xc832514220)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc829682068, 0xc8326edac0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc829682068, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:264 +0x3d
net/http/httptest.(*historyListener).Accept(0xc821000390, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/httptest/server.go:48 +0x63
net/http.(*Server).Serve(0xc831719b60, 0x7f8df9d45fe8, 0xc821000390, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1887 +0xb3
created by net/http/httptest.(*Server).Start
	/usr/local/go/src/net/http/httptest/server.go:109 +0x380

goroutine 102758 [select]:
github.com/coreos/etcd/rafthttp.(*pipeline).handle(0xc822a53680)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:91 +0x615
created by github.com/coreos/etcd/rafthttp.newPipeline
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/pipeline.go:77 +0x205

goroutine 102781 [select]:
net/http.(*persistConn).writeLoop(0xc821d4d6b0)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 100915 [select]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).run(0xc8327701a0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:571 +0x612
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).start
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:433 +0x49a

goroutine 102746 [select]:
github.com/coreos/etcd/etcdserver/etcdhttp.capabilityLoop(0xc820158000)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/etcdhttp/capability.go:68 +0x1ad
created by github.com/coreos/etcd/etcdserver/etcdhttp.NewClientHandler
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/etcdhttp/client.go:63 +0x6b

goroutine 102726 [select]:
net/http.(*persistConn).writeLoop(0xc820ffc210)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d
exit status 2
FAIL	github.com/coreos/etcd/integration	383.495s
```

--- a/raft/node.go
+++ b/raft/node.go
@@ -160,7 +160,6 @@ func StartNode(c *Config, peers []Peer) Node {
 	// TODO(bdarnell): These entries are still unstable; do we need to preserve
 	// the invariant that committed < unstable?
 	r.raftLog.committed = r.raftLog.lastIndex()
-	r.Commit = r.raftLog.committed
 	// Now apply them, mainly so that the application can call Campaign
 	// immediately after StartNode in tests. Note that these nodes will
 	// be added to raft twice: here and when the application's Ready
@@ -453,8 +452,8 @@ func newReady(r *raft, prevSoftSt *SoftState, prevHardSt pb.HardState) Ready {
 	if softSt := r.softState(); !softSt.equal(prevSoftSt) {
 		rd.SoftState = softSt
 	}
-	if !isHardStateEqual(r.HardState, prevHardSt) {
-		rd.HardState = r.HardState
+	if hardSt := r.hardState(); !isHardStateEqual(hardSt, prevHardSt) {
+		rd.HardState = hardSt
 	}
 	if r.raftLog.unstable.snapshot != nil {
 		rd.Snapshot = *r.raftLog.unstable.snapshot

--- a/raft/raft.go
+++ b/raft/raft.go
@@ -138,9 +138,10 @@ func (c *Config) validate() error {
 }
 
 type raft struct {
-	pb.HardState
-
 	id uint64
+
+	Term uint64
+	Vote uint64
 
 	// the log
 	raftLog *raftLog
@@ -239,6 +240,14 @@ func (r *raft) hasLeader() bool { return r.lead != None }
 
 func (r *raft) softState() *SoftState { return &SoftState{Lead: r.lead, RaftState: r.state} }
 
+func (r *raft) hardState() pb.HardState {
+	return pb.HardState{
+		Term:   r.Term,
+		Vote:   r.Vote,
+		Commit: r.raftLog.committed,
+	}
+}
+
 func (r *raft) quorum() int { return len(r.prs)/2 + 1 }
 
 func (r *raft) nodes() []uint64 {
@@ -295,7 +304,7 @@ func (r *raft) sendAppend(to uint64) {
 		m.Snapshot = snapshot
 		sindex, sterm := snapshot.Metadata.Index, snapshot.Metadata.Term
 		r.logger.Debugf("%x [firstindex: %d, commit: %d] sent snapshot[index: %d, term: %d] to %x [%s]",
-			r.id, r.raftLog.firstIndex(), r.Commit, sindex, sterm, to, pr)
+			r.id, r.raftLog.firstIndex(), r.raftLog.committed, sindex, sterm, to, pr)
 		pr.becomeSnapshot(sindex)
 		r.logger.Debugf("%x paused sending replication messages to %x [%s]", r.id, to, pr)
 	} else {
@@ -525,7 +534,6 @@ func (r *raft) Step(m pb.Message) error {
 		if r.state != StateLeader {
 			r.logger.Infof("%x is starting a new election at term %d", r.id, r.Term)
 			r.campaign()
-			r.Commit = r.raftLog.committed
 		} else {
 			r.logger.Debugf("%x ignoring MsgHup because already leader", r.id)
 		}
@@ -550,7 +558,6 @@ func (r *raft) Step(m pb.Message) error {
 		return nil
 	}
 	r.step(r, m)
-	r.Commit = r.raftLog.committed
 	return nil
 }
 
@@ -742,8 +749,8 @@ func stepFollower(r *raft, m pb.Message) {
 }
 
 func (r *raft) handleAppendEntries(m pb.Message) {
-	if m.Index < r.Commit {
-		r.send(pb.Message{To: m.From, Type: pb.MsgAppResp, Index: r.Commit})
+	if m.Index < r.raftLog.committed {
+		r.send(pb.Message{To: m.From, Type: pb.MsgAppResp, Index: r.raftLog.committed})
 		return
 	}
 
@@ -765,11 +772,11 @@ func (r *raft) handleSnapshot(m pb.Message) {
 	sindex, sterm := m.Snapshot.Metadata.Index, m.Snapshot.Metadata.Term
 	if r.restore(m.Snapshot) {
 		r.logger.Infof("%x [commit: %d] restored snapshot [index: %d, term: %d]",
-			r.id, r.Commit, sindex, sterm)
+			r.id, r.raftLog.committed, sindex, sterm)
 		r.send(pb.Message{To: m.From, Type: pb.MsgAppResp, Index: r.raftLog.lastIndex()})
 	} else {
 		r.logger.Infof("%x [commit: %d] ignored snapshot [index: %d, term: %d]",
-			r.id, r.Commit, sindex, sterm)
+			r.id, r.raftLog.committed, sindex, sterm)
 		r.send(pb.Message{To: m.From, Type: pb.MsgAppResp, Index: r.raftLog.committed})
 	}
 }
@@ -782,13 +789,13 @@ func (r *raft) restore(s pb.Snapshot) bool {
 	}
 	if r.raftLog.matchTerm(s.Metadata.Index, s.Metadata.Term) {
 		r.logger.Infof("%x [commit: %d, lastindex: %d, lastterm: %d] fast-forwarded commit to snapshot [index: %d, term: %d]",
-			r.id, r.Commit, r.raftLog.lastIndex(), r.raftLog.lastTerm(), s.Metadata.Index, s.Metadata.Term)
+			r.id, r.raftLog.committed, r.raftLog.lastIndex(), r.raftLog.lastTerm(), s.Metadata.Index, s.Metadata.Term)
 		r.raftLog.commitTo(s.Metadata.Index)
 		return false
 	}
 
 	r.logger.Infof("%x [commit: %d, lastindex: %d, lastterm: %d] starts to restore snapshot [index: %d, term: %d]",
-		r.id, r.Commit, r.raftLog.lastIndex(), r.raftLog.lastTerm(), s.Metadata.Index, s.Metadata.Term)
+		r.id, r.raftLog.committed, r.raftLog.lastIndex(), r.raftLog.lastTerm(), s.Metadata.Index, s.Metadata.Term)
 
 	r.raftLog.restore(s)
 	r.prs = make(map[uint64]*Progress)
@@ -848,7 +855,6 @@ func (r *raft) loadState(state pb.HardState) {
 	r.raftLog.committed = state.Commit
 	r.Term = state.Term
 	r.Vote = state.Vote
-	r.Commit = state.Commit
 }
 
 // isElectionTimeout returns true if r.elapsed is greater than the

--- a/raft/raft_test.go
+++ b/raft/raft_test.go
@@ -957,8 +957,8 @@ func TestMsgAppRespWaitReset(t *testing.T) {
 		Type:  pb.MsgAppResp,
 		Index: 1,
 	})
-	if sm.Commit != 1 {
-		t.Fatalf("expected Commit to be 1, got %d", sm.Commit)
+	if sm.raftLog.committed != 1 {
+		t.Fatalf("expected committed to be 1, got %d", sm.raftLog.committed)
 	}
 	// Also consume the MsgApp messages that update Commit on the followers.
 	sm.readMessages()
@@ -1046,7 +1046,7 @@ func TestRecvMsgVote(t *testing.T) {
 		case StateLeader:
 			sm.step = stepLeader
 		}
-		sm.HardState = pb.HardState{Vote: tt.voteFor}
+		sm.Vote = tt.voteFor
 		sm.raftLog = &raftLog{
 			storage:  &MemoryStorage{ents: []pb.Entry{{}, {Index: 1, Term: 2}, {Index: 2, Term: 2}}},
 			unstable: unstable{offset: 3},

--- a/raft/rawnode.go
+++ b/raft/rawnode.go
@@ -105,7 +105,7 @@ func NewRawNode(config *Config, peers []Peer) (*RawNode, error) {
 	}
 	// Set the initial hard and soft states after performing all initialization.
 	rn.prevSoftSt = r.softState()
-	rn.prevHardSt = r.HardState
+	rn.prevHardSt = r.hardState()
 
 	return rn, nil
 }
@@ -191,7 +191,7 @@ func (rn *RawNode) HasReady() bool {
 	if !r.softState().equal(rn.prevSoftSt) {
 		return true
 	}
-	if !IsEmptyHardState(r.HardState) && !isHardStateEqual(r.HardState, rn.prevHardSt) {
+	if hardSt := r.hardState(); !IsEmptyHardState(hardSt) && !isHardStateEqual(hardSt, rn.prevHardSt) {
 		return true
 	}
 	if r.raftLog.unstable.snapshot != nil && !IsEmptySnap(*r.raftLog.unstable.snapshot) {

--- a/raft/status.go
+++ b/raft/status.go
@@ -33,7 +33,7 @@ type Status struct {
 // getStatus gets a copy of the current raft status.
 func getStatus(r *raft) Status {
 	s := Status{ID: r.id}
-	s.HardState = r.HardState
+	s.HardState = r.hardState()
 	s.SoftState = *r.softState()
 
 	s.Applied = r.raftLog.applied

